algoliasearch/index.go

```diff
@@ -770,3 +759,87 @@ func (i *index) SearchRulesWithRequestOptions(params Map, opts *RequestOptions)
 	err = i.client.request(&res, "POST", path, params, read, opts)
 	return
 }
+
+// ReplaceAllSynonyms replace all the synonyms of the current index with the given ones.
+func (i *index) ReplaceAllSynonyms(synonyms []Synonym) (res UpdateTaskRes, err error) {
+	return i.ReplaceAllSynonymsWithRequestOptions(synonyms, nil)
+}
+
+// ReplaceAllSynonymsWithRequestOptions is the same as ReplaceAllSynonyms but it also
+// accepts extra RequestOptions.
+func (i *index) ReplaceAllSynonymsWithRequestOptions(synonyms []Synonym, opts *RequestOptions) (res UpdateTaskRes, err error) {
+	return i.BatchSynonymsWithRequestOptions(synonyms, true, true, opts)
+}
+
+// ReplaceAllRules replace all the rules of the current index with the given ones.
+func (i *index) ReplaceAllRules(rules []Rule) (res BatchRulesRes, err error) {
+	return i.ReplaceAllRulesWithRequestOptions(rules, nil)
+}
+
+// ReplaceAllRulesWithRequestOptions is the same as ReplaceAllRules but it also
+// accepts extra RequestOptions.
+func (i *index) ReplaceAllRulesWithRequestOptions(rules []Rule, opts *RequestOptions) (res BatchRulesRes, err error) {
+	return i.BatchRulesWithRequestOptions(rules, true, true, opts)
+}
+
+// ReplaceAllObjects replace all the objects of the current index with the given ones.
+func (i *index) ReplaceAllObjects(objects []Object) error {
+	return i.ReplaceAllObjectsWithRequestOptions(objects, nil)
+}
+
+// ReplaceAllObjectsWithRequestOptions is the same as ReplaceAllObjects but it also
+// accepts extra RequestOptions.
+func (i *index) ReplaceAllObjectsWithRequestOptions(objects []Object, opts *RequestOptions) (err error) {
+	tmpIndexName := fmt.Sprintf("%s_tmp_%d", i.name, time.Now().Nanosecond())
+
+	defer func() {
+		if err != nil {
+			i.client.DeleteIndex(tmpIndexName)
+		}
+	}()
+
+	var taskIDs []int
+
+	tmpIndex := i.client.InitIndex(tmpIndexName)
+	batchSize := 1000
+
+	for i := 0; i < len(objects); i += batchSize {
+		j := i + batchSize
+		if j > len(objects) {
```

Comment:
Performance issue ( at least on PHP ): You may want to increase a counter instead of calculating the size of the objects in every iteration. Example:
```python

        batch = []
        batch_size = 1000
        count = 0
        for obj in objects:
            batch.append(obj)
            count += 1

            if count == batch_size:
                response = tmp_index.save_objects(batch, request_options)
                responses.append(response)
                batch = []
                count = 0

        if batch:
            response = tmp_index.save_objects(batch, request_options)
            responses.append(response)
```

Reply:
(not sure if the code snippet you share do match the comment you wrote)

Retrieving the length of a *slice* in Go (vector-like in other runtimes) is O(1) so we're fine here.